### ISSUE TITLE
KeyError: 'USER'

### ISSUE THREAD

**Issue by jwettenhall** [reecetech/django_informixdb_vault#6]
I've seen a few of these errors via Sentry:

```
KeyError: 'USER'
```

See traceback below.

Maybe multiple threads are trying to perform the `del self.settings_dict['USER']` around the same time?

The code triggering the exception is here: https://github.com/reecetech/django_informixdb_vault/blob/master/django_informixdb_vault/base.py#L172

I suppose it couldn't do any harm to wrap it in a try / catch?

```
    try:
        del self.settings_dict['USER']
    except KeyError:
        pass

    try:
        del self.settings_dict['PASSWORD']
    except KeyError:
        pass
```

or use a threading lock?

```
    import threading
    ...
    self.lock = threading.Lock()

    ...

    if self.lock.acquire(blocking=False):
        del self.settings_dict['USER']
        del self.settings_dict['PASSWORD']
```

Traceback:
```
KeyError: 'USER'
  File "rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "rest_framework/generics.py", line 252, in get
    return self.retrieve(request, *args, **kwargs)
  File "rest_framework/mixins.py", line 54, in retrieve
    instance = self.get_object()
  File "rest_framework/generics.py", line 96, in get_object
    obj = get_object_or_404(queryset, **filter_kwargs)
  File "rest_framework/generics.py", line 19, in get_object_or_404
    return _get_object_or_404(queryset, *filter_args, **filter_kwargs)
  File "django/shortcuts.py", line 93, in get_object_or_404
    return queryset.get(*args, **kwargs)
  File "django/db/models/query.py", line 402, in get
    num = len(clone)
  File "django/db/models/query.py", line 256, in __len__
    self._fetch_all()
  File "django/db/models/query.py", line 1242, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "django/db/models/query.py", line 55, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
  File "django/db/models/sql/compiler.py", line 1140, in execute_sql
    cursor = self.connection.cursor()
  File "django/db/backends/base/base.py", line 256, in cursor
    return self._cursor()
  File "django/db/backends/base/base.py", line 233, in _cursor
    self.ensure_connection()
  File "django/db/backends/base/base.py", line 217, in ensure_connection
    self.connect()
  File "django/db/backends/base/base.py", line 194, in connect
    conn_params = self.get_connection_params()
  File "django_informixdb_vault/base.py", line 172, in get_connection_params
    del self.settings_dict['USER']
```

**Comment by ps-jay** [comment]
Either approach looks fine to me!  Thanks for looking into it @jwettenhall

**Comment by ps-jay** [comment]
Resolved in #7
